src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1180,6 +1188,9 @@ class run_impl
 
     virtual void
     set_arg_value(const argument& arg, const arg_range<uint32_t>& value) = 0;
+
+    virtual void
+    set_sk_args(const uint8_t* args, const uint32_t size) = 0;
```

Comment:
I don't think this is necessary.  The current set_arg_value should do just fine.  If not it should be set_arg_value with a arg_range instantiated for uint8.